planner/core/plan_cost.go

```diff
@@ -172,7 +174,37 @@ func (p *PhysicalTableReader) GetPlanCost(taskType property.TaskType) (float64,
 
 // GetPlanCost calculates the cost of the plan if it has not been calculated yet and returns the cost.
 func (p *PhysicalIndexMergeReader) GetPlanCost(taskType property.TaskType) (float64, error) {
-	return 0, errors.New("not implemented")
+	if p.planCostInit {
+		return p.planCost, nil
+	}
+	p.planCost = 0
+	netFactor := p.ctx.GetSessionVars().GetNetworkFactor(nil)
+	if tblScan := p.tablePlan; tblScan != nil {
+		childCost, err := tblScan.GetPlanCost(property.CopSingleReadTaskType)
+		if err != nil {
+			return 0, err
+		}
+		p.planCost += childCost // child's cost
+		tblStats := getTblStats(tblScan)
+		rowSize := tblStats.GetAvgRowSize(p.ctx, tblScan.Schema().Columns, false, false)
```

Comment:
Here we also ignore the table-side seek cost for the same reason as https://github.com/pingcap/tidb/pull/34091#discussion_r855007584 (ranges is nil, it is not built before the execution), right?

Reply:
Actually, I'm not sure about this. I guess it is just because the original author of `IndexMergeReader` forgot to accumulate this part of costs... For safety and to keep compatible with the old implementation, I won't fix it in this PR and I'll add some comments.

Reply:
In the old implementation, the table-side seek cost is also calculated in
https://github.com/pingcap/tidb/blob/b7f26143e500379fec94c286badcac26bdfc90d6/planner/core/task.go#L1075-L1081
However, the table-side seek cost is 0 since `ts.Ranges` is empty. I think it's ok that we don't calculate the table-side seek cost currently since we don't have a good method to measure it. Maybe we can consider how to measure it in the future.